warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -319,51 +337,70 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.START_OFFSET),
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.END_OFFSET));
+            }
+            try {
+                excerptIterator.init(source, excerptIteratorOptions, env);
+                excerptIterator.setHitTermsList(hitTermValues);
+                excerptIterator.setDirection(excerptFields.getDirection(field).toUpperCase().trim());
+                excerptIterator.setOrigHalfSize(origHalfSize);
+                if (i != 0) {
+                    excerptIterator.setTrim(true);
+                }
+                excerptIterator.seek(range, Collections.emptyList(), false);
+                // if an excerpt is returned...
+                if (excerptIterator.hasTop()) {
+                    Key topKey = excerptIterator.getTopKey();
+                    String[] parts = topKey.getColumnQualifier().toString().split(Constants.NULL);
+                    // The column qualifier is expected to be field\0phraseWithScores\0phraseWithoutScores\0oneBestExcerpt.
+                    if (parts.length != 4) {
+                        log.warn("{} returned top key with incorrectly-formatted column qualifier in key: {} when scanning for excerpt [{},{}] for field {} within range {}",
+                                        TermFrequencyExcerptIterator.class.getSimpleName(), topKey, start, end, field, range);
+                        return null;
+                    }
 
-        // return phrase based on direction
-        String result = String.join(" ", hitPhrase); // if no direction given, return everything
-        String direction = excerptFields.getDirection(phraseParts[0]).toUpperCase().trim();
-        if (direction.equals(BEFORE)) { // remove tokens prior to hit term
-            result = StringUtils.substringBeforeLast(result, "]") + "]";
-        } else if (direction.equals(AFTER)) { // remove tokens after hit term
-            result = "[" + StringUtils.substringAfter(result, "[");
+                    // if we have reached the limit of times to try, or we have no stop words removed
+                    if (i == 1 || !parts[1].equals("XXXWESKIPPEDAWORDXXX")) {
+                        return parts[1] + Constants.NULL + parts[2] + Constants.NULL + parts[3];
+                    }
+                } else {
+                    return null;
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to scan for excerpt [" + start + "," + end + "] for field " + field + " within range " + range, e);
+            }
+            i++;
         }
-
-        return result;
+        // it should always return from inside the loop so if this is reached something went very wrong
+        throw new RuntimeException("This should never be reached. Something went wrong!");
```

Comment:
Perhaps we return null here after we've exhausted all of our attempts instead of returning null from within the loop.

Reply:
as with the other two comments, no more returning null as having a blank excerpt with no message tells us nothing. 

Will output the message "something went wrong generating your excerpt"